src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultVisualStudioOpenDocumentManager.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Export(typeof(VisualStudioOpenDocumentManager))]
+    internal class DefaultVisualStudioOpenDocumentManager : VisualStudioOpenDocumentManager
+    {
+        private readonly IVsFileChangeEx _fileChangeService;
+        private readonly IVsRunningDocumentTable _runningDocumentTable;
+        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private readonly ErrorReporter _errorReporter;
+
+        private List<VisualStudioDocumentTracker> _documents;
+        private Dictionary<string, ViewImportChangeTracker> _viewImportChangeTrackerCache;
+
+        [ImportingConstructor]
+        public DefaultVisualStudioOpenDocumentManager(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            [Import(typeof(VisualStudioWorkspace))] Workspace workspace,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            if (workspace == null)
+            {
+                throw new ArgumentNullException(nameof(workspace));
+            }
+
+            if (editorAdaptersFactoryService == null)
+            {
+                throw new ArgumentNullException(nameof(editorAdaptersFactoryService));
+            }
+
+            _fileChangeService = serviceProvider.GetService(typeof(SVsFileChangeEx)) as IVsFileChangeEx;
+            _runningDocumentTable = serviceProvider.GetService(typeof(SVsRunningDocumentTable)) as IVsRunningDocumentTable;
+            _editorAdaptersFactoryService = editorAdaptersFactoryService;
+
+            _documents = new List<VisualStudioDocumentTracker>();
+            _viewImportChangeTrackerCache = new Dictionary<string, ViewImportChangeTracker>(StringComparer.OrdinalIgnoreCase);
```

Comment:
We should not hardcode this everywhere, we should start writing all of our code assuming xplat. It's just a good habit.